R/export_cdisc.R

```diff
@@ -93,19 +100,38 @@ export_cdisc <- function(res_nca) {
     #  Recode PPTESTCD PKNCA names to CDISC abbreviations
     mutate(
       PPTESTCD = translate_terms(PPTESTCD, mapping_col = "PKNCA", target_col = "PPTESTCD"),
+      PPTEST = translate_terms(PPTESTCD, mapping_col = "PPTESTCD", target_col = "PPTEST"),
       DOMAIN = "PP",
       # Group ID
-      PPGRPID =  paste(ANALYTE, PCSPEC, paste("CYCLE", DOSNO,  sep = " "), sep = "-"),
+      PPGRPID = {
+        if ("PCGRPID" %in% names(.)) PCGRPID
+        else if ("AVISIT" %in% names(.)) paste(ANALYTE, DRUG, PCSPEC, AVISIT, sep = "-")
+        else paste(DRUG, PCSPEC, DOSNO, sep = "-")
+      },
       # Parameter Category
       PPCAT = if ("PARAM" %in% names(.)) PARAM else ANALYTE,
       PPSCAT = "NON-COMPARTMENTAL",
       PPDOSNO = DOSNO,
       PPSPEC = PCSPEC,
       # Specific ID variables
-      PPSPID = "TBD",
-      # TODO Results in Standard Units if ORRESU is not in standard units
-      PPSTRESN = as.numeric(PPSTRES),
-      PPSTRESC = as.character(PPSTRES),
+      PPSPID = {
+        if ("PCSPID" %in% names(.)) PCSPID
+        else if ("EXSPID" %in% names(.)) EXSPID
+        else NA
+      },
```

Comment:
```suggestion
      PPSPID = PPSPID = paste0("/F:EDT-", STUDYID, "_PKPARAM_aNCA"),
```
Per requirements of PP,
PPSPID = "Concatenate "/F:EDT-" with the dataset/analysis name [PREPP.PPDSNAME]"

Reply:
I checked and there is nothing about PCSPID and EXSPID, these IDs are very different from what we want